labgrid/driver/power/netio_kshell.py

```diff
@@ -1,37 +1,46 @@
 """ tested with NETIO 4C, should be compatible with all NETIO 4-models """
 
 import re
-import telnetlib
+
+import pexpect
 
 PORT = 1234
 
 def power_set(host, port, index, value):
     index = int(index)
     assert 1 <= index <= 4
     value = "1" if value else "0"
-    tn = telnetlib.Telnet(host, port, 1)
-    tn.read_until(b"\r\n", 0.5)
-    tn.write(b"login admin admin\n")
-    tn.read_until(b"250 OK\r\n", 0.5)
-    tn.write(f"port {index} {value}\n".encode())
-    tn.read_until(b"250 OK\r\n", 0.5)
-    tn.write(b"quit\n")
-    tn.close()
+
+    with pexpect.spawn(f"telnet {host} {port}", timeout=1) as tn:
```

Comment:
This timeout now applies to any `expect()` call, instead of only the connection creation.

Reply:
Correct, that's intentional. Prior to this, all calls had their own timeout. I think it's more readable this way.